src/main/java/org/cru/godtools/api/notifications/NotificationResource.java

```diff
@@ -0,0 +1,100 @@
+package org.cru.godtools.api.notifications;
+
+import com.google.common.base.Optional;
+import org.cru.godtools.api.authorization.AuthorizationResource;
+import org.cru.godtools.domain.authentication.AuthorizationRecord;
+import org.cru.godtools.domain.authentication.AuthorizationService;
+import org.cru.godtools.domain.authentication.UnauthorizedException;
+import org.cru.godtools.domain.notifications.Device;
+import org.cru.godtools.domain.notifications.DeviceService;
+import org.cru.godtools.domain.notifications.Notification;
+import org.cru.godtools.domain.notifications.NotificationService;
+import org.cru.godtools.domain.properties.GodToolsProperties;
+import org.cru.godtools.domain.properties.GodToolsPropertiesFactory;
+import org.jboss.logging.Logger;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.UUID;
+
+/**
+ * Created by matthewfrederick on 12/30/14.
+ */
+@Path("/notification")
+public class NotificationResource
+{
+	@Inject
+	DeviceService deviceService;
+	@Inject
+	AuthorizationService authorizationService;
+	@Inject
+	NotificationService notificationService;
+
+	private final GodToolsProperties properties = new GodToolsPropertiesFactory().get();
+
+	Logger log = Logger.getLogger(AuthorizationResource.class);
+
+	@POST
+	@Path("/{registrationId}")
+	public Response registerDevice(@PathParam("registrationId")String registrationId,
+								   @HeaderParam("deviceId") String deviceIdHeader, @QueryParam("deviceId") String deviceIdParam)
+	{
+		String id = deviceIdHeader == null ? deviceIdParam : deviceIdHeader;
+
+		log.info("Registering device: " + id + " with registrationId: " + registrationId);
+
+		Device device = new Device();
+		device.setId(UUID.randomUUID());
+		device.setRegistrationId(registrationId);
+		device.setDeviceId(id);
+
+		deviceService.insert(device);
+
+		return Response.ok().build();
+	}
+
+	@POST
```

Comment:
don't change this now b/c it will require an update to the Android app which I don't want to do right now, but updates should be a PUT, not a POST.  it's working as it is, so we can clean this up later.